lib/engines/hckinstall/setup_scripts_helper.rb

```diff
@@ -23,9 +23,15 @@ def validate_setup_scripts_config(config)
       raise(AutoHCKError, "Undefined HLK setup scripts configs: #{extra_keys.join(', ')}.")
     end
 
-    def download_kit_installer(url, kit, hck_setup_scripts_path)
+    def download_kit_installer(url, kit, hck_setup_scripts_path, iso)
       dw = Downloader.new(@logger)
-      dw.download(url, "#{hck_setup_scripts_path}/Kits/#{kit}Setup.exe")
+      if iso
+        iso_path = "#{hck_setup_scripts_path}/Kits/#{kit}Setup.iso"
+        dw.download(url, iso_path) unless File.exist?(iso_path)
```

Comment:
Why do you need to check `File.exist?(iso_path)` when you don't if `iso` is false?

Reply:
We don't want to download a big ISO every time. If the installation fails near after the ISO download we will spend a lot of time just for downloading. HLK online installer (exe) is several MB, so redownloading it is no so a long process.

Reply:
`prepare_setup_scripts_config` already has a `File.exist?` check so let's move it before `download_kit_installer` to see if HLK is downloaded, whether it is iso or not.